src/module/system/damage/iwr.ts

```diff
@@ -225,6 +248,38 @@ function applyIWR(actor: ActorPF2e, roll: Rolled<DamageRoll>, rollOptions: Set<s
     return { finalDamage, applications, persistent };
 }
 
+/**
+ * Find a resistance "redirection" among a list of candidates: that is, one that matches the `highest` resistance type
+ * and would result is less damage being resisted.
+ */
+function getResistanceRedirection({
+    resistances,
+    highest,
+    candidates,
+}: {
+    resistances: WorkingResistanceData[];
+    highest: WorkingResistanceData;
+    candidates: RedirectedResistance[];
+}): { to: WorkingResistanceData | null; redirect: RedirectedResistance } | null {
+    return candidates
+        .filter((rr) => [highest.type, null].includes(rr.from) && rr.to !== highest.type)
+        .reduce(
```

Comment:
Would this function be clearer if the comparison was made relative to `const highestValue = Math.max(highest.value, mostReduction?.to?.value ?? 0);`, returning mostReduction (even if its null) if not?